libs/prelude/Prelude/Interfaces.idr

```diff
@@ -105,6 +105,37 @@ public export
 ignore : Functor f => f a -> f ()
 ignore = map (const ())
 
+||| Bifunctors
+||| @f The action of the Bifunctor on pairs of objects
+public export
+interface Bifunctor f where
+  ||| The action of the Bifunctor on pairs of morphisms
+  |||
+  ||| ````idris example
+  ||| bimap (\x => x + 1) reverse (1, "hello") == (2, "olleh")
+  ||| ````
+  |||
+  bimap : (a -> c) -> (b -> d) -> f a b -> f c d
+  bimap f g = fmap f . smap g
+
+  ||| The action of the Bifunctor on morphisms pertaining to the first object
+  |||
+  ||| ````idris example
+  ||| fmap (\x => x + 1) (1, "hello") == (2, "hello")
+  ||| ````
+  |||
+  fmap : (a -> c) -> f a b -> f c b
```

Comment:
This is confusing as `fmap` is usually a `Functor` method in Haskell.
I would prefer e.g. `mapLeft` and `mapRight` respectively.
In Agda's stdlib we use `map₁` and `map₂` but the explicit reference to
sides might be clearer (& is also a direct generalisation of the corresponding
functions on `Either`).

I would also be happy with `first` and `second` as they are established
names in Haskell AFAIK.